src/sentry/api/endpoints/auth_index.py

```diff
@@ -96,7 +97,7 @@ def _validate_superuser(self, validator, request):
         validator.is_valid()
         authenticated = None
 
-        if (
+        if is_self_hosted() and (
             request.user.has_usable_password()
             or Authenticator.objects.filter(user_id=request.user.id, type=3).exists()
         ):
```

Comment:
This could use a little explanation. Maybe we could extract the "if" expression out to a helper function (returning a boolean) and use a docstring to explain the motivation.

That would also be a good opportunity to replace `type=3` with a constant.